src/test/java/com/accesspoint/rulesengine/apitests/ruleset/RulesetTest.java

```diff
@@ -466,4 +468,94 @@ public void givenRuleset_whenPOSTRulesConditionsIsEmpty_thenCustomErrorIsCalled(
+        List<Condition> fakeCondition1 = new ArrayList<>();
+        fakeCondition1.add(fakeCon1_1);
+        List<Condition> fakeCondition2 = new ArrayList<>();
+        fakeCondition2.add(fakeCon2_1);
+        fakeCondition2.add(fakeCon2_2);
+        Rule fakeRule1 = new Rule(100L, 11, FORWARD, null, null);
+        Rule fakeRule2 = new Rule(200L, 0, FORWARD, null, fakeCondition2);
+        List<Rule> fakeRules = new ArrayList<>();
+        fakeRules.add(fakeRule1);
+        fakeRules.add(fakeRule2);
+        Ruleset ruleset =
+                Ruleset.builder()
+                        .name("Test1")
+                        .rules(fakeRules)
+                        .id(100L)
+                        .creation_date(Timestamp.valueOf("2000-01-01 01:15:30.500"))
+                        .build();
+
+        // Mock ALL repository methods that get called in the service
+
+        when(rulesetRepository.save(Mockito.any(Ruleset.class))).thenReturn(ruleset);
+        when(rulesetRepository.getReferenceById(Mockito.any(Long.class))).thenReturn(ruleset);
+        when(ruleRepository.save(Mockito.eq(fakeRule1))).thenReturn(fakeRule1);
+        when(conditionRepository.save(Mockito.eq(fakeCon1_1))).thenReturn(fakeCon1_1);
+        when(rulesetRepository.findById(Mockito.eq(100L))).thenReturn(Optional.ofNullable(ruleset));
+
+        // Given the ruleset, when post request, then response body is as expected
+        given()
+                .contentType(ContentType.JSON)
+                .body(ruleset)
+                .when()
+                .delete("/ruleset/100")
+                .then()
+                .assertThat()
+                .statusCode(204);
+    }
+    // TODO: if id doesn't exist, throw error
+    @Test
+    public void givenRuleset_whenDELETERulesetwithInvalidID_thenCustomErrorIsThrown() {
+
+        // Building out the mock ruleset
+        Condition fakeCon1_1 = null;
+        Condition fakeCon2_1 = new Condition(200L, FRONT, EMPTY, null);
+        Condition fakeCon2_2 = new Condition(201L, LEFT, EMPTY, null);
+        List<Condition> fakeCondition1 = new ArrayList<>();
+        fakeCondition1.add(fakeCon1_1);
+        List<Condition> fakeCondition2 = new ArrayList<>();
+        fakeCondition2.add(fakeCon2_1);
+        fakeCondition2.add(fakeCon2_2);
+        Rule fakeRule1 = new Rule(100L, 11, FORWARD, null, null);
+        Rule fakeRule2 = new Rule(200L, 0, FORWARD, null, fakeCondition2);
+        List<Rule> fakeRules = new ArrayList<>();
+        fakeRules.add(fakeRule1);
+        fakeRules.add(fakeRule2);
+        Ruleset ruleset =
+                Ruleset.builder()
+                        .name("Test1")
+                        .rules(fakeRules)
+                        .id(100L)
+                        .creation_date(Timestamp.valueOf("2000-01-01 01:15:30.500"))
+                        .build();
+
+        // Mock ALL repository methods that get called in the service
```

Comment:
Same thing about the repository mocking, you should only be mocking what is called as everything else is just excess code.